torch/csrc/autograd/python_engine.cpp

```diff
@@ -91,27 +105,11 @@ std::shared_ptr<FutureVariableList> PythonEngine::execute_with_graph_task(
 
 PyObject *THPEngineClass = nullptr;
 
-static bool _reinitialize_engine = false;
-
-static void _maybe_reinitialize_engine_after_fork() {
-  // This is "probably" thread-safe because the flag is set in a fork handler
-  // before any threads are created, and this function is only called with the
-  // GIL held. However, using fork + threads is playing with fire so this is
-  // more of a "best effort" thing. For example, if the fork occurs while the
-  // backwards threads hold a lock, we'll probably deadlock in the engine
-  // destructor.
-  if (_reinitialize_engine) {
-    engine.~PythonEngine();
-    new (&engine) torch::autograd::python::PythonEngine();
-    _reinitialize_engine = false;
-  }
-}
-
 // Implementation of torch._C._EngineBase.run_backward
 PyObject *THPEngine_run_backward(THPEngine *self, PyObject *args, PyObject *kwargs)
 {
   HANDLE_TH_ERRORS
-  _maybe_reinitialize_engine_after_fork();
+  python::PythonEngine::get_python_engine();
```

Comment:
This is not needed anymore now that you properly use `get_python_engine()` below to get the engine right?

Reply:
Good point, let me update the diff